server/vulnerabilities/nvd/cve.go

```diff
@@ -272,3 +285,142 @@ func checkCVEs(
 
 	return foundVulns, nil
 }
+
+// Returns the versionEndExcluding string for the given CVE and host software meta
+// data, if it exists in the NVD feed.  This effectively gives us the version of the
+// software it needs to upgrade to in order to address the CVE.
+func getMatchingVersionEndExcluding(ctx context.Context, cve string, hostSoftwareMeta *wfn.Attributes, dict cvefeed.Dictionary) (string, error) {
+	vuln, ok := dict[cve].(*feednvd.Vuln)
+	if !ok {
+		return "", nil
+	}
+
+	// Schema() maps to the JSON schema of the NVD feed for a given CVE
+	vulnSchema := vuln.Schema()
+	if vulnSchema == nil {
+		return "", nil
+	}
+
+	config := vulnSchema.Configurations
+	if config == nil {
+		return "", nil
+	}
+
+	nodes := config.Nodes
+	if len(nodes) == 0 {
+		return "", nil
+	}
+
+	cpeMatch := findCPEMatch(nodes)
+	if cpeMatch == nil || len(cpeMatch) == 0 {
+		return "", nil
+	}
+
+	// convert the host software version to semver for later comparison
+	formattedVersion := preprocessVersion(wfn.StripSlashes(hostSoftwareMeta.Version))
+	softwareVersion, err := semver.NewVersion(formattedVersion)
+	if err != nil {
+		return "", ctxerr.Wrap(ctx, err, "parsing software version", hostSoftwareMeta.Product, hostSoftwareMeta.Version)
+	}
+
+	// Check if the host software version matches any of the CPEMatch rules.
+	// CPEMatch rules can include version strings for the following:
+	// - versionStartIncluding
+	// - versionStartExcluding
+	// - versionEndExcluding
+	// - versionEndIncluding - not used in this function
```

Comment:
Curious as to why are we are not using `versionEndIncluding` (some entries have only `versionEndIncluding`)

Reply:
Ah, probably because we cannot say which version resolves it :)

Let's add a small comment about that, for future readers.